src/bin/wd_copier.rs

```diff
@@ -12,10 +15,40 @@ struct Opt {
     dst: PathBuf,
 }
 
+fn get_dirty_bytes() -> u64 {
+    let meminfo = procfs::Meminfo::new().unwrap();
+    meminfo.dirty
+}
+
+fn sync_progress_bar(
+    rx: mpsc::Receiver<()>,
+    mut progress_bar: progress::Bar,
+    starting_dirty: u64,
+    dirty_after_copy: u64,
+) {
+    progress_bar.set_job_title("syncing... (2/2)");
+    loop {
+        let percent = 100
+            - ((get_dirty_bytes().saturating_sub(starting_dirty))
+                / (dirty_after_copy / 100));
+        progress_bar.reach_percent(percent as i32);
+        thread::sleep(Duration::from_millis(500));
+        match rx.try_recv() {
+            Ok(_) | Err(mpsc::TryRecvError::Disconnected) => {
+                break;
+            }
+            Err(mpsc::TryRecvError::Empty) => {}
+        }
```

Comment:
I think this match could be simplified, maybe:

```rust
if matches!(rx.try_recv(), Ok(_) | Err(mpsc::TryRecvError::Disconnected) {
  return;
}
```

Reply:
Good call, I didn't know about `matches!`

78ecdc8